cnf-certification-test/networking/suite.go

```diff
@@ -83,41 +84,42 @@ var _ = ginkgo.Describe(common.NetworkingTestKey, func() {
 //nolint:funlen
 func testListenAndDeclared(env *provider.TestEnvironment) {
 	var k declaredandlistening.Key
-	failedContainers := []string{}
-	for _, cut := range env.Containers {
+	var failedPods []*v1.Pod
+	for _, podUnderTest := range env.Pods {
 		declaredPorts := make(map[declaredandlistening.Key]bool)
 		listeningPorts := make(map[declaredandlistening.Key]bool)
-		ports := cut.Data.Ports
-		logrus.Debugf("%s declaredPorts: %v", cut.StringShort(), ports)
-		for j := 0; j < len(ports); j++ {
-			k.Port = int(ports[j].ContainerPort)
-			k.Protocol = string(ports[j].Protocol)
-			declaredPorts[k] = true
-		}
-		outStr, errStr, err := crclient.ExecCommandContainerNSEnter(cmd, cut, env)
-		if err != nil || errStr != "" {
-			tnf.ClaimFilePrintf("Failed to execute command %s on %s, err: %s, errStr: %s", cmd, cut.StringShort(), err, errStr)
-			failedContainers = append(failedContainers, cut.StringShort())
-			continue
-		}
-		declaredandlistening.ParseListening(outStr, listeningPorts)
-		if len(listeningPorts) == 0 {
-			tnf.ClaimFilePrintf("%s does not have any listening ports.", cut.StringShort())
-			continue
-		}
-		// compare between declaredPort,listeningPort
-		undeclaredPorts := declaredandlistening.CheckIfListenIsDeclared(listeningPorts, declaredPorts)
-		for k := range undeclaredPorts {
-			tnf.ClaimFilePrintf("The port %d on protocol %s not declared on %s", k.Port, k.Protocol, cut.StringShort())
-		}
-		if len(undeclaredPorts) != 0 {
-			failedContainers = append(failedContainers, fmt.Sprintf("%s undeclaredPorts: %v", cut.StringShort(), undeclaredPorts))
+		for _, container := range podUnderTest.Spec.Containers {
+			ports := container.Ports
+			logrus.Debugf("container %s (%s) declaredPorts: %v", container.Name, provider.PodToString(podUnderTest), ports)
+			for j := 0; j < len(ports); j++ {
+				k.Port = int(ports[j].ContainerPort)
+				k.Protocol = string(ports[j].Protocol)
+				declaredPorts[k] = true
+			}
+			firstPodContainer := &podUnderTest.Spec.Containers[0]
```

Comment:
The for loop getting the declared ports of the containers of the pod should finish right before the "firstPodContainer" variable declaration. Please review my previous comment again.